src/Discord/Builders/Components/Section.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is a part of the DiscordPHP project.
+ *
+ * Copyright (c) 2015-present David Cole <david.cole1340@gmail.com>
+ *
+ * This file is subject to the MIT license that is bundled
+ * with this source code in the LICENSE.md file.
+ */
+
+namespace Discord\Builders\Components;
+
+/**
+ * Section components allow you to define up to 3 text display components and add either a thumbnail or button to the right side.
+ *
+ * @link https://discord.com/developers/docs/interactions/message-components#section
+ *
+ * @since 10.4.0
+ */
+class Section extends Component implements Contracts\ComponentV2
+{
+    /**
+     * Array of text display components.
+     *
+     * @var TextDisplay[]
+     */
+    private $components = [];
+
+    /**
+     * Accessory component (Thumbnail or Button).
+     *
+     * @var Thumbnail|Button|null
+     */
+    private $accessory;
+
+    /**
+     * Creates a new section.
+     *
+     * @return self
+     */
+    public static function new(): self
+    {
+        return new self();
+    }
+
+    /**
+     * Adds a text display component to the section.
+     * Text displays can only be used within sections.
+     *
+     * @param Component $component Text display component to add.
+     *
+     * @throws \InvalidArgumentException Component is not a TextDisplay.
+     * @throws \OverflowException Section exceeds 3 text components.
+     *
+     * @return $this
+     */
+    public function addComponent(Component $component): self
+    {
+        if (! ($component instanceof TextDisplay)) {
+            throw new \InvalidArgumentException('Section can only contain TextDisplay components. Use setAccessory() for Thumbnail or Button.');
+        }
+
+        if (count($this->components) >= 3) {
+            throw new \OverflowException('You can only have 3 text components per section.');
+        }
+
+        $this->components[] = $component;
+
+        return $this;
+    }
+
+    /**
+     * Sets the accessory component of the section.
+     * Only Thumbnail or Button components can be used as accessories.
+     *
+     * @param Thumbnail|Button $component Thumbnail or Button component.
+     *
+     * @throws \InvalidArgumentException Component is not a Thumbnail or Button.
+     *
+     * @return $this
+     */
+    public function setAccessory(Thumbnail|Button $component): self
+    {
+        $this->accessory = $component;
```

Comment:
The spec is telling us to not hard code the supported components yet (anyway leave the phpdoc as is)

```suggestion
    public function setAccessory(Component $component): self
    {
        if (! ($component instanceof Thumbnail || $component instanceof Button)) {
            trigger_error('Accessory may only contain Thumbnail or Button component.', E_USER_NOTICE);
        }
        $this->accessory = $component;
```

Reply:
No, throwing error is the same as hardcoding it, the spec tells us NOT to

if you insist users should be informed then go back with my previous suggestions of `trigger_error()` as the `E_USER_NOTICE` doest not throw error but showing notice tracing to this code in their logs.

Reply:
The rest of the library does not do this. It should be a separate PR.